spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java

```diff
@@ -235,7 +236,19 @@ public String getId() {
 
 		@Override
 		public DeploymentState getState() {
-			return isAlive(process) ? DeploymentState.deployed : DeploymentState.failed;
+			boolean alive = isAlive(process);
+			if (!alive) {
+				return DeploymentState.failed;
+			}
+			try {
+				HttpURLConnection urlConnection = (HttpURLConnection) moduleUrl.openConnection();
+				urlConnection.connect();
+				urlConnection.disconnect();

```

Comment:
Should the similar approach on handling the exception on `disconnect` in the InProcessModuleDeployer be followed here as well?